src/utils/copyProps.js

```diff
@@ -13,8 +17,10 @@ export default function copyProps(props, transformFn = identity) {
   const copied = {}
 
   for (const prop in props) {
+    /* istanbul ignore else */
     if (props.hasOwnProperty(prop)) {
-      if (typeof prop === 'object') {
+      if (typeof props[prop] === 'object') {
```

Comment:
What about using the `isObject` util here and convert the if to a ternary operator?

Reply:
That might work as well.

Reply:
changed to use `isObject` + ternary